server/lib/subscriptions.js

```diff
@@ -49,6 +51,13 @@ function hasReachedQuantity(order) {
   return order.Subscription.chargeNumber !== null && order.Subscription.chargeNumber === order.Subscription.quantity;
 }
 
+function getOrderPlan(order) {
+  if (order.collective.slug === PLANS_COLLECTIVE_SLUG && order.Tier.data) {
```

Comment:
Like in `subscribeOrUpgradePlan`, you should check that the slug is in in `plans`, see `plans[newPlan]`